packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -259,6 +261,32 @@ class WebFs {
         return Response.ok(file.readAsBytesSync(), headers: <String, String>{
           'Content-Type': 'text/javascript',
         });
+      } else if (request.url.path.endsWith('part.js')) {
+        // Lazily unpack any deferred imports in release/profile mode. These are
+        // placed into an archive by build_runner, and are named based on the main
+        // entrypoint + a "part" suffix (Though the actual names are arbitrary).
+        // To make this easier to deal with they are copied into a temp directory.
+        if (partFiles == null) {
+          final File dart2jsArchive = fs.file(fs.path.join(
+              flutterProject.dartTool.path,
+              'build',
+              'flutter_web',
+              '${flutterProject.manifest.appName}',
+              'lib',
+              '${targetBaseName}_web_entrypoint.dart.js.tar.gz'));
+          if (dart2jsArchive.existsSync()) {
+            final Archive archive = TarDecoder().decodeBytes(dart2jsArchive.readAsBytesSync());
+            partFiles = fs.systemTempDirectory.createTempSync('_flutter_tool')
```

Comment:
Can this overflow temp disk space if not cleaned up explicitly?

Reply:
Made this into a class with a dispose method